account/src/main/java/com/caldevsupplychain/account/controller/AccountController.java

```diff
@@ -0,0 +1,262 @@
+import com.caldevsupplychain.notification.mail.service.EmailService;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/account/v1")
+@AllArgsConstructor
+public class AccountController {
+
+	private ModelMapper modelMapper;
+
+	private AccountService accountService;
+
+	private AccountWSBuilder accountWSBuilder;
+
+	/* validators */
+	private SignupValidator signupValidator;
+	private LoginValidator loginValidator;
+	private EditUserValidator editUserValidator;
+
+	private EmailService emailService;
+
+	// use validator to customize error code and message
+	@InitBinder("userSignupDTO")
+	public void initUserSignupFormBinder(WebDataBinder binder) {binder.addValidators(signupValidator);}
+	@InitBinder("userLoginDTO")
+	public void initUserLoginFormBinder(WebDataBinder binder) {binder.addValidators(loginValidator);}
+	@InitBinder("userEditDTO")
+	public void initUserEditFormBinder(WebDataBinder binder) {binder.addValidators(editUserValidator);}
+
+
+
+	/************************************************************************************************
+	 |									Account API													|
+	 ************************************************************************************************/
+
+
+	@GetMapping("/demo")
+	public ResponseEntity<?> demo (HttpServletRequest request) throws NullPointerException {
+
+		throw new NullPointerException("Throwing from demo");
+//        throw new IllegalAccessError("Illegal throw from dog");
+//        return null;
+	}
+
+
+	@PostMapping("/signup")
+	public ResponseEntity<?> signup (
+			@RequestParam(required = false, defaultValue = "USER") String role,
+			@Validated @RequestBody UserSignupDTO userSignupDTO,
+			HttpServletRequest request
+	) throws MessagingException
+	{
+		// convert back to UserWS after signup validation
+		UserWS userWS = modelMapper.map(userSignupDTO, UserWS.class);
+
+		// need to update after activated link, and user select specific role
+		userWS.setRole(role);
+
+		/*  TODO: Signup Process Thought
+			if we break down the signup process to 3 parts
+			1. user sign up (username, email, password) << currently this user only have USER role
+			2. send email activation link
+			3. user activated -> navigate to another detail signup form:
+				  - where user can select role ( << after selected, frontEnd UI change to specific form for either
+					Designer <> Agent <> Manufacturer to signup for more details information
+			4. user complete  all the signup process:
+				  - send notification to admin for review then to approve/unapprove these users before users
+					can go into a specific dashboard view
+		  */
+
+		// check if user pre-exist or not
+		if (accountService.userExist(userWS.getEmailAddress())) {
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_EXIST.name(), "Account already registered."), HttpStatus.CONFLICT);
+		}
+
+		// success on checking, create user
+		User user = accountService.createUser(userWS);
+
+		log.warn("CREATE USER SUCCESS");
```

Comment:
Use 'log.debug' since this is not a warning message.

Reply:
fixed